piperider_cli/initializer.py

```diff
@@ -110,3 +115,100 @@ def exec(dbt_project_path=None, dbt_profiles_dir=None):
+                        source = ds.credential['dbname']
+                        source_type = 'Database'
+                    elif ds.type_name in ['snowflake']:
+                        source = ds.credential['database']
+                        source_type = 'Database'
+                    elif ds.type_name in ['bigquery']:
+                        source = ds.credential['dataset']
+                        source_type = 'Dataset'
+                    list_table.add_row(ds.type_name, ds.name, source_type, source)
+
+                layout_table = Table(
+                    title='PipeRider Configuration',
+                    title_style='bold magenta',
+                    show_header=False,
+                    show_edge=True,
+                    box=box.SIMPLE_HEAVY)
+                layout_table.add_column("List")
+                layout_table.add_column("Yaml", width=80)
+                layout_table.add_row(list_table, yaml_markdown)
+
+            console.print(layout_table)
+        else:
+            console.print('[bold red]Piperider workspace does not exist[/bold red] ')
+
+    @staticmethod
+    def delete(report_dir=None):
+        console = Console()
+
+        config = Configuration.load()
+
+        if FANCY_USER_INPUT:
+            questions = [
+                inquirer.List('datasource',
+                              message='Which datasource do you want to delete?',
+                              choices=[(ds.name, ds) for ds in
+                                       config.dataSources]),
+                inquirer.Confirm('confirm',
+                                 message='Are you sure?')
+            ]
+            answers = inquirer.prompt(questions)
+        else:
+            console.print('[[yellow]?[/yellow]] Which datasource do you want to delete?')
+            idx = 0
+            for ds in config.dataSources:
+                console.print(f'    [[green]{idx}[/green]] {ds.name}')
+                idx += 1
+            answer = Prompt.ask('Please enter the number of delete datasource:', choices=[str(i) for i in range(idx)])
+            confirm = Prompt.ask('Are you sure? [y/n]', choices=['y', 'n'], default='N')
+            answers = {'datasource': config.dataSources[int(answer)], 'confirm': confirm == 'y'}
+
+        if answers['confirm'] is False:
+            console.print('[bold red]Abort to delete datasource[/bold red]')
+            return
+
+        config.delete_datasource(answers['datasource'])
+        console.rule('Datasource deleted')
+
+    @staticmethod
+    def add(report_dir=None):
+        console = Console()
+        config = Configuration.load()
+        console.rule('Add datasource')
+        cls, name = DataSource.ask()
+        ds: DataSource = cls(name=name)
+        config.dataSources.append(ds)
+        if _ask_user_update_credentials(ds):
+            _generate_piperider_workspace()
+            config.dump(PIPERIDER_CONFIG_PATH)
```

Comment:
the same issue as above due to `config.dump` implementation

Reply:
change to use method `flush_datasource` instead of `dump`